community-tutorials/ruby-on-rails-in-production/01-en.md

```diff
@@ -0,0 +1,269 @@
+
+```sh
+sudo service nginx start
+```
+
+If we now access the IP or address of the server via Webbrowser and receive the message "Welcome to NGINX", we have configured NGINX correctly.
+
+
+Now we can create a new configuration file by creating the following file with the editor nano:
+
+```sh
+sudo nano /etc/nginx/sites-enabled/rails_app
+```
+
+We copy the following content into it:
+
+```
+server {
+  listen 80;
+  listen [::]:80;
+
+  server_name YOUR_DOMAIN;
+  root /var/www/rails_app/public;
+
+  passenger_enabled on;
+  passenger_app_env production;
+
+  location /cable {
+    passenger_app_group_name myapp_websocket;
+    passenger_force_max_concurrent_requests_per_process 0;
+  }
+
+  client_max_body_size 10m;
+
+  location ~ ^/(assets|packs) {
+    expires max;
+    gzip_static on;
+  }
+}
+```
+
+You can modify the following details according to your needs:
+
+* server_name
+  * Instead of the `YOUR_DOMAIN`, you can enter your domain here on which your Rails app should run.
+* root
+  * `/var/www/rails_app` is the location of our Rails app. You can customise this, the important thing is that the path always ends with `/public`.
+* passenger_app_env
+  * If you prefer to run your Rails app for development in the development environment, you can specify `development` here instead of `production`.
+* location /cable
+  * If you are sure that your app will not use an action cable for websockets, you can remove this block from the file.
+* client_max_body_size
+  * If you want to allow your users to upload files to your Rails app, you can set a maximum upload file size here. In the default setting, this is 10 MB.
+* location ~ ^/(assets|packs)
+  * The assets and packs directories contain javascript and css files from your Rails app. To make these files more performant, they are cached and zipped with the block. You can extend this with more directories.
+
+
+Once you have saved the file, we change to the directory `/var/www/`. Now you can copy your existing Rails app into the folder `rails_app`. Alternatively, you can use tools like Capistrano for deployment.
+
+To install Rails, use the gem install command provided by RubyGems:
+
+```sh
+gem install rails
+```
+
+If you want to start from scratch, you can create a new rails app here with the name `rails_app` and execute the following command:
+
+```sh
+rails new rails_app
+```
+
+After creating the project folder, rails new runs bundle install to install the necessary gems and then yarn to install the necessary JS dependencies. We can then switch to the project folder using `cd rails_app` and start development. Happy Coding ;) 
+
+Sqlite3 was selected by default as the database, which is sufficient for small applications in most cases. If you want to do more, install PostgreSQL or another database on the server and configure it in your Rails app.
+
+
+To finally compile the javascript files execute the following command:
+
+```sh
+RAILS_ENV=production rails assets:precompile
```

Comment:
I can´t run this code with rails_user.

Reply:
I think you have not installed rbenv as rails_user. That is why these errors occur.